### PR TITLE
Scheduled Tasks to Jellyfin.Api

### PR DESCRIPTION
**Pull request by crobibero**
Changes
Migrated the ScheduledTasksService.cs from MediaBrowser.Api to Jellyfin.Api which is using ASP.NET WebAPI.

Issues
Part of #2872

### CODE REVIEW

File: Jellyfin.Api/Controllers/ScheduledTasksController.cs [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+                    });
+                }
+
+                var taskInfos = tasks.Select(ScheduledTaskHelpers.GetTaskInfo);
+
+                return Ok(taskInfos);
+            }
+            catch (Exception e)
+            {
+                return StatusCode(StatusCodes.Status500InternalServerError, e.Message);
```
Comment (cvium) [inline Jellyfin.Api/Controllers/ScheduledTasksController.cs:84]: Exception middleware or MVC exception filter. I prefer the former though.

Comment (crobibero) [reply Jellyfin.Api/Controllers/ScheduledTasksController.cs:84]: I’ll create a PR for an exception middleware, sounds like something I should have learned about a while ago

File: Jellyfin.Api/Controllers/ScheduledTasksController.cs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        /// Get tasks.
+        /// </summary>
+        /// <param name="isHidden">Optional filter tasks that are hidden, or not.</param>
+        /// <param name="isEnabled">Optional filter tasks that are enabled, or not.</param>
+        /// <response code="200">Scheduled tasks retrieved.</response>
+        /// <returns>The list of scheduled tasks.</returns>
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public IEnumerable<IScheduledTaskWorker> GetTasks(
+            [FromQuery] bool? isHidden = false,
```
Comment (cvium) [inline Jellyfin.Api/Controllers/ScheduledTasksController.cs:41]: Can this be null?

File: Jellyfin.Api/Controllers/ScheduledTasksController.cs [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+            var task = _taskManager.ScheduledTasks.FirstOrDefault(i =>
+                string.Equals(i.Id, taskId, StringComparison.OrdinalIgnoreCase));
+
+            if (task == null)
+            {
+                return NotFound();
+            }
+
+            var result = ScheduledTaskHelpers.GetTaskInfo(task);
+            return Ok(result);
```
Comment (cvium) [inline Jellyfin.Api/Controllers/ScheduledTasksController.cs:86]: Should be without `Ok()`, right?

File: Jellyfin.Api/Controllers/ScheduledTasksController.cs [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+
+namespace Jellyfin.Api.Controllers
+{
+    /// <summary>
+    /// Scheduled Tasks Controller.
+    /// </summary>
+    // [Authenticated]
```
Comment (cvium) [inline Jellyfin.Api/Controllers/ScheduledTasksController.cs:17]: Remove?

File: Jellyfin.Api/Controllers/ScheduledTasksController.cs (hunk 2 above)
Comment (crobibero) [reply Jellyfin.Api/Controllers/ScheduledTasksController.cs:42]: it can, if the variable isn't given it we don't filter by it

Comment (cvium) [reply Jellyfin.Api/Controllers/ScheduledTasksController.cs:42]: But it has a default value?

Comment (crobibero) [reply Jellyfin.Api/Controllers/ScheduledTasksController.cs:42]: And this is why we review :) Fixed in latest commit

File: Jellyfin.Api/Controllers/ScheduledTasksController.cs [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+            var task = _taskManager.ScheduledTasks.FirstOrDefault(o =>
+                o.Id.Equals(taskId, StringComparison.OrdinalIgnoreCase));
+
+            if (task == null)
+            {
+                return NotFound();
+            }
+
+            _taskManager.Execute(task, new TaskOptions());
+            return Ok();
```
Comment (Ullmie02) [inline Jellyfin.Api/Controllers/ScheduledTasksController.cs:108]:
```suggestion
            return NoContent();
```

File: Jellyfin.Api/Controllers/ScheduledTasksController.cs [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+        {
+            var task = _taskManager.ScheduledTasks.FirstOrDefault(o =>
+                o.Id.Equals(taskId, StringComparison.OrdinalIgnoreCase));
+            if (task == null)
+            {
+                return NotFound();
+            }
+
+            task.Triggers = triggerInfos;
+            return Ok();
```
Comment (Ullmie02) [inline Jellyfin.Api/Controllers/ScheduledTasksController.cs:158]:
```suggestion
            return NoContent();
```

File: Jellyfin.Api/Controllers/ScheduledTasksController.cs [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+            var task = _taskManager.ScheduledTasks.FirstOrDefault(o =>
+                o.Id.Equals(taskId, StringComparison.OrdinalIgnoreCase));
+
+            if (task == null)
+            {
+                return NotFound();
+            }
+
+            _taskManager.Cancel(task);
+            return Ok();
```
Comment (Ullmie02) [inline Jellyfin.Api/Controllers/ScheduledTasksController.cs:132]:
```suggestion
            return NoContent();
```

**Pull request merged**